src/main/groovy/org/cadixdev/gradle/licenser/Licenser.groovy

```diff
@@ -53,66 +58,101 @@ class Licenser implements Plugin<Project> {
+                        def extensionIgnoreFailures = extension.ignoreFailures
+                        def extensionLineEnding = extension.lineEnding
+                        createAndroidTask(CHECK_TASK, LicenseCheck, headers, set) {
+                            ignoreFailures.set(extensionIgnoreFailures)
+                        }
+                        createAndroidTask(FORMAT_TASK, LicenseUpdate, headers, set) {
+                            lineEnding.set(extensionLineEnding)
+                        }
+                    }
                 }
             }
 
-            def globalCheck = task(CHECK_TASK + 's')
-            task('licenseCheck', dependsOn: globalCheck)
-            def globalFormat = task(FORMAT_TASK + 's')
-            task('licenseFormat', dependsOn: globalFormat)
-
-            afterEvaluate {
-                project.tasks.findByName(CHECK_TASK_NAME)?.dependsOn globalCheck
-            }
-
-            // Wait a bit until creating the tasks
-            afterEvaluate {
-                def headers = []
-                extension.conditionalProperties.reverseEach {
-                    headers << prepareHeader(extension, it)
+            extension.tasks.all { LicenseTaskProperties props ->
+                def extensionIgnoreFailures = extension.ignoreFailures
+                def extensionLineEnding = extension.lineEnding
+                createCustomTask(CHECK_TASK, LicenseCheck, props) {
+                    ignoreFailures.set(extensionIgnoreFailures)
                 }
-                headers << prepareHeader(extension, extension)
-
-                extension.sourceSets.each {
-                    def check = createTask(CHECK_TASK, LicenseCheck, headers, it)
-                    check.ignoreFailures = extension.ignoreFailures
-                    globalCheck.dependsOn check
-                    globalFormat.dependsOn createTask(FORMAT_TASK, LicenseUpdate, headers, it)
+                createCustomTask(FORMAT_TASK, LicenseUpdate, props) {
+                    lineEnding.set(extensionLineEnding)
                 }
+            }
 
-                extension.androidSourceSets.each {
-                    def check = createAndroidTask(CHECK_TASK, LicenseCheck, headers, it)
-                    check.ignoreFailures = extension.ignoreFailures
-                    globalCheck.dependsOn check
-                    globalFormat.dependsOn createAndroidTask(FORMAT_TASK, LicenseUpdate, headers, it)
-                }
+            // Then configure catch-all tasks
+            def globalCheck = tasks.register(CHECK_TASK + 's') {
+                dependsOn(tasks.withType(LicenseCheck))
+            }
+            tasks.register('licenseCheck') {
+                dependsOn(globalCheck)
+            }
+            def globalFormat = tasks.register(FORMAT_TASK + 's') {
+                dependsOn(tasks.withType(LicenseUpdate))
+            }
+            tasks.register('licenseFormat') {
+                group = FORMATTING_GROUP
```

Comment:
It's this `formatting` group a common identifier? Is there even a standard on this?

Reply:
It's an identifier used by the `ktlint` plugin, though it seems to not be universal since `spotless` uses the `verification` group